packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -540,21 +540,24 @@ async function fluid({ file, args = {}, reporter }) {
     options.sizes = `(max-width: ${presentationWidth}px) 100vw, ${presentationWidth}px`
   }
 
-  // Create sizes (in width) for the image. If the max width of the container
-  // for the rendered markdown file is 800px, the sizes would then be: 200,
-  // 400, 800, 1200, 1600, 2400.
+  // Create sizes (in width) for the image if no custom sizes are provided. If
+  // the max width of the container for the rendered markdown file is 800px,
+  // the sizes would then be: 200, 400, 800, 1200, 1600, 2400.
   //
   // This is enough sizes to provide close to the optimal image size for every
   // device size / screen resolution while (hopefully) not requiring too much
   // image processing time (Sharp has optimizations thankfully for creating
   // multiple sizes of the same input file)
-  const fluidSizes = []
-  fluidSizes.push(options[fixedDimension] / 4)
-  fluidSizes.push(options[fixedDimension] / 2)
-  fluidSizes.push(options[fixedDimension])
-  fluidSizes.push(options[fixedDimension] * 1.5)
-  fluidSizes.push(options[fixedDimension] * 2)
-  fluidSizes.push(options[fixedDimension] * 3)
+  const fluidSizes = !options.customSizes || !options.customSizes.length
+    ? [
+      options[fixedDimension] / 4,
+      options[fixedDimension] / 2,
+      options[fixedDimension],
+      options[fixedDimension] * 1.5,
+      options[fixedDimension] * 2,
+      options[fixedDimension] * 3,
+    ]
+    : options.customSizes.slice()
```

Comment:
do we need to create copy of array?

Reply:
Nope. Turns out I was just being overly safe. I'll get rid of it in a future commit :)